### PR TITLE
Add GPAD8-2R (gpad8_2r) Macro Pad under gkeyboard

### PR DESCRIPTION
**Pull request by gkeyboard**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/gkeyboard/gpad8_2r/config.h [hunk 1]
```diff
@@ -0,0 +1,73 @@
+// Copyright 2023 gkeyboard (@gkeyboard)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "config_common.h"
+
```
Comment (waffle87) [inline keyboards/gkeyboard/gpad8_2r/config.h:7]:
```suggestion
```

File: keyboards/gkeyboard/gpad8_2r/config.h [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+#define DYNAMIC_KEYMAP_LAYER_COUNT 8
+
+#define RGB_DI_PIN GP19
+#define RGB_MATRIX_LED_COUNT 16
+#define DRIVER_LED_TOTAL 16
+#define RGBLED_NUM 16
```
Comment (waffle87) [inline keyboards/gkeyboard/gpad8_2r/config.h:13]:
```suggestion
```

File: keyboards/gkeyboard/gpad8_2r/config.h [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* Bootmagic Lite key configuration */
+#define BOOTMAGIC_LITE_ROW 0
+#define BOOTMAGIC_LITE_COLUMN 0
```
Comment (waffle87) [inline keyboards/gkeyboard/gpad8_2r/config.h:73]:
```suggestion
```

File: keyboards/gkeyboard/gpad8_2r/keymaps/via/readme.md [hunk 4]
```diff
@@ -0,0 +1 @@
+# Via keymap for GPAD8-2R
```
Comment (waffle87) [inline keyboards/gkeyboard/gpad8_2r/keymaps/via/readme.md:1]: This file should be removed.

File: keyboards/gkeyboard/gpad8_2r/readme.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# GPAD8-2R
+
+![GPAD8-2R](https://i.imgur.com/jAaO7gY.png)
```
Comment (waffle87) [inline keyboards/gkeyboard/gpad8_2r/readme.md:3]:
```suggestion
![GPAD8-2R](https://i.imgur.com/jAaO7gYh.png)
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/gkeyboard/gpad8_2r/keymaps/via/readme.md (hunk 4 above)
Comment (gkeyboard) [reply keyboards/gkeyboard/gpad8_2r/keymaps/via/readme.md:1]: Thank you.

**Review by waffle87 — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**